[Ryan-Reno] iP

Ryan-Reno opened the pull request:
LITE

Lite is the next generation's ToDo List! It's :
- text-based
- ~~EASY~~ SUPER EASY to learn and use

All you need to do is,
1. download it from [here](https://github.com/Ryan-Reno/ip).
2. double-click it.
3. add your tasks.
4. let it keep track of all the tasks for you 😃 

And it is **COMPLETELY FREE!**

Features :
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon) 

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```
public static void main(String[] args) {
        Chatbot chatbot = new Chatbot();
        chatbot.start();
 }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
Renamed the chatbot

Commit:
Added greeting and exits automatically

Commit:
Added echo

Commit:
Added a list that stores tasks

Commit:
Added markings for tasks

Commit:
Added todo, deadline, and event

Commit:
Added automatic Text UI Testing

Commit:
Added exception handling for invalid inputs

Commit:
Added delete task functionality

AjayArvind2207 commented on src/main/java/Chatbot.java:
```diff
@@ -0,0 +1,208 @@
+        }
+
+        if (instruction[0].equals("delete")) {
+            this.delete(instruction);
```
Consider removing the 'this' reference since there is inconsistent styling with addToDo among other instructions
```suggestion
            delete(instruction);
```

AjayArvind2207 commented on src/main/java/Chatbot.java:
```diff
@@ -0,0 +1,208 @@
+public class Chatbot {
+    private Scanner scanner = new Scanner(System.in);
+    private String currentInput;
+    private ArrayList<Task> list;
```
Suggestion: Why not have a Tasks class to encapsulate the ArrayList<Task> objects? That way it makes the code in Chatbot.java a lot less dense, and you can abstract away most of the complexity into that class.

Ryan-Reno replied:
Sorry I don't quite get what you mean. Isn't the point of the Task class to encapsulate the ArrayList objects? If so, doesn't my Task class behave the same with the Tasks class you suggested?

AjayArvind2207 commented on src/main/java/Lite.java:
```diff
@@ -0,0 +1,15 @@
+public class Lite {
+    public static void main(String[] args) {
+        String logo = "  LLLLL      IIIIIIIII  TTTTTTTTTT  EEEEEEEEEE  \n" +
```
This is super cool and I really enjoy it :D

AjayArvind2207 reviewed (commented):
Code looks good and well written. I did feel as though you could have abstracted a bit more in terms of classes -- using smaller classes to encapsulate behaviour you have used in Chatbot.java.

Ryan-Reno committed:
Add save mechanism for task list

Ryan-Reno committed:
Merge branch 'branch-level-7'

Ryan-Reno committed:
Abstracted more classes to adhere with OOP principles

Ryan-Reno committed:
Divide classes into packages

Ryan-Reno committed:
Merge branch 'add-gradle-support'

Ryan-Reno committed:
Add JUnit testing for Event and Task

Ryan-Reno committed:
Add JavaDocs

Ryan-Reno committed:
Merge branch 'branch-A-JavaDoc'

Ryan-Reno committed:
Followed coding standards

Ryan-Reno committed:
Add find functionality to find a task by a keyword

Ryan-Reno committed:
Merge branch 'branch-A-CodingStandard'

Ryan-Reno committed:
Merge branch 'branch-Level-9'

jovantanyk commented on src/main/java/lite/Ui.java:
Method Names could be clearer. parse() could be renamed into executeCommand() instead. 
Methods like find, delete, mark addDeadline, addEvent could be renamed into findTask, deleteTask, markTaskAsDone, addDeadlineTask, addEventTask for more clarity.
Overall not a bad attempt however more OOP could be used. The Ui class is currently parsing input strings, processing business logic, and handling output all at the same time. You could separate the command parsing into a different class and allow the Ui class to solely handle text generation

jovantanyk commented on src/main/java/lite/util/Printer.java:
Method names seem inconsistent. toDoException could be renamed to printToDoException. successfulFind and unsuccessfulFind could be renamed to printFound() and printNotFound(). Since the Printer class solely handles output, methods should have consistent naming.
Also, it seems redundant to have methods to print Exceptions here when you're also handling exceptions in LiteExceptions.

Ryan-Reno replied:
Thanks for this feedback! Was not thinking properly when creating the methods and simply put it within the Printer class.

jovantanyk commented on src/main/java/lite/task/TaskList.java:
Method names could be better. remove() could be more specific like removeTask(). taskString() is fine but toString() would be better.

jovantanyk reviewed (commented):
Overall Good attempt but method names could use more consistent naming. UI class is currently handling too many functions and could be split up more. Exception handling is a bit weird as it doesn't actually catch any exceptions. LiteException class should extend Exception so you can actually use try catch blocks to handle exceptions.

jovantanyk commented on src/main/java/lite/Ui.java:
```diff
@@ -43,12 +45,33 @@ public boolean parse(TaskList tasks) {
         return false;
     }
 
+    private void find (String instruction[], TaskList tasks) {
```
method name could be clearer for example findTask() instead of just find()

jovantanyk commented on src/main/java/lite/task/TaskList.java:
```diff
@@ -59,6 +68,15 @@ public void outputTasks() {
+    /**
+     * Outputs the string representation of the TaskList
+     */
+    public void taskString() {
```
toString() would be better and follow naming conventions

Ryan-Reno committed:
Task stores date and time instead of string

Ryan-Reno committed:
Refactor methods to adhere with OOP

Ryan-Reno committed:
Change EventTest to fit with new output

Ryan-Reno committed:
Implement GUI with JavaFX and FXML

Ryan-Reno committed:
Merge branch 'branch-Level-10'

Ryan-Reno committed:
Use assert in Ui to ensure important assumptions

There are several methods in Ui where tasks will be added and deleted.
As a result, the size of the tasklist will go up and down accordingly.

The usage of assert will ensure that the size of these will adjust
according to what task has been presented, whether it is a delete or
add.

Ryan-Reno committed:
Improve code quality by creating Parser class

The parsing method within Ui is put into its own class called Parser.
Its task is to receive the input and determine which task the input is
associated with

Ryan-Reno committed:
Merge pull request #1 from Ryan-Reno/branch-A-CodeQuality

Branch a code quality

Ryan-Reno committed:
Detect duplicate tasks

Added a constraint that does not allow tasks that has been added to be
added again into the Task List. Tasks with different types (Event, Todo,
and Deadline) with the same description will be considered a different
task.

Two deadline tasks will also be considered different if their deadlines
do not end at the same time.

Ryan-Reno committed:
Merge pull request #2 from Ryan-Reno/branch-BCD-Extension

Detect duplicate tasks

Ryan-Reno committed:
Fix DialogBox height limit and add bye command

Ryan-Reno committed:
Merge branch 'branch-BCD-Extension'

Ryan-Reno committed:
Add JavaDocs to methods

Ryan-Reno committed:
Add User Guide

Ryan-Reno committed:
Add Ui.png to the user guide

Ryan-Reno committed:
Add picture to the website

Ryan-Reno committed:
Add greetings when first opening the app
